src/main/java/com/mongodb/quickstart/Update.java

```diff
@@ -0,0 +1,76 @@
+package com.mongodb.quickstart;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.FindOneAndUpdateOptions;
+import com.mongodb.client.model.ReturnDocument;
+import com.mongodb.client.model.UpdateOptions;
+import com.mongodb.client.result.UpdateResult;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonWriterSettings;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Updates.*;
+
+public class Update {
+
+    public static void main(String[] args) {
+        Logger.getLogger("org.mongodb.driver").setLevel(Level.ALL);
+        JsonWriterSettings prettyPrint = JsonWriterSettings.builder().indent(true).build();
+
+        try (MongoClient mongoClient = MongoClients.create(System.getProperty("mongodb.uri"))) {
+            MongoDatabase sampleTrainingDB = mongoClient.getDatabase("sample_training");
+            MongoCollection<Document> gradesCollection = sampleTrainingDB.getCollection("grades");
+
+            // update one document
+            Bson filter = eq("student_id", 10000);
+            Bson updateOperation = set("comment", "You should learn MongoDB!");
+            UpdateResult updateResult = gradesCollection.updateOne(filter, updateOperation);
+            System.out.println("=> Updating the doc with {\"student_id\":10000}. Adding comment.");
+            System.out.println(gradesCollection.find(filter).first().toJson(prettyPrint));
+            System.out.println(updateResult);
+
+            // upsert
+            filter = and(eq("student_id", 10002d), eq("class_id", 10d));
```

Comment:
Why use a double here? Elsewhere student_id is an int32 eg: why `10002d` and not just `10002`

Reply:
It's actually a double in all the documents in the Atlas data sample for some reason. So I'm using a double again here to "stick to the plan".